mod-dcb/src/main/resources/volaris/mod-dcb/features/cancelling-flow-for-borrower-flow.feature

```diff
@@ -0,0 +1,240 @@
+
+
+    * def updateToAwaitingPickupRequest = read('classpath:volaris/mod-dcb/features/samples/transaction/update-dcb-transaction-to-awaiting-pickup.json')
+
+    Given path 'transactions' , transactionId , 'status'
+    And request updateToAwaitingPickupRequest
+    When method PUT
+    Then status 200
+
+    Given path 'transactions' , transactionId , 'status'
+    When method GET
+    Then status 200
+    And match $.status == 'AWAITING_PICKUP'
+    And match $.role == 'BORROWER'
+
+    Given path 'transactions' , transactionId , 'status'
+    And request updateToCancelRequest
+    When method PUT
+    Then status 200
+
+    Given path 'transactions' , transactionId , 'status'
+    When method GET
+    Then status 200
+    And match $.status == 'CANCELLED'
+    And match $.role == 'BORROWER'
+
+    Given path 'request-storage', 'requests'
+    Given param query = '(item.barcode= ' + '3X' + ' and itemId = ' + id1 + ' )'
+    When method GET
+    Then status 200
+    And match $.totalRecords == 1
+    And match $.requests[0].status == 'Closed - Cancelled'
+
+  #Negative
+  Scenario: Cancel DCB Transaction manually after ITEM_CHECKED_OUT
+    * def transactionId = '400'
+    * def id1 = 'a9b73276-77b6-11ee-b962-0242ac120006'
+    * def createTransaction = call read('classpath:volaris/mod-dcb/reusable/create-dcb-transaction-for-borrower.feature') { transactionId: '#(transactionId)', extItemId: '#(id1)', itemBarcode: '4X' }
+    * def updateToCancelRequest = read('classpath:volaris/mod-dcb/features/samples/transaction/update-dcb-transaction-to-cancel.json')
+
+    * def updateToAwaitingPickupRequest = read('classpath:volaris/mod-dcb/features/samples/transaction/update-dcb-transaction-to-awaiting-pickup.json')
+
+    Given path 'transactions' , transactionId , 'status'
+    And request updateToAwaitingPickupRequest
+    When method PUT
+    Then status 200
+
+    Given path 'transactions' , transactionId , 'status'
+    When method GET
+    Then status 200
+    And match $.status == 'AWAITING_PICKUP'
+    And match $.role == 'BORROWER'
+
+    * def updateToItemCheckedOutRequest = read('classpath:volaris/mod-dcb/features/samples/transaction/update-dcb-transaction-to-item-check-out.json')
+
+    Given path 'transactions' , transactionId , 'status'
+    And request updateToItemCheckedOutRequest
+    When method PUT
+    Then status 200
+
+    Given path 'transactions' , transactionId , 'status'
+    When method GET
+    Then status 200
+    And match $.status == 'ITEM_CHECKED_OUT'
+    And match $.role == 'BORROWER'
+
+    Given path 'transactions' , transactionId , 'status'
+    And request updateToCancelRequest
+    When method PUT
+    Then status 400
+    And match $.errors[0].message == 'Cannot cancel transaction dcbTransactionId: 400. Transaction already in status: ITEM_CHECKED_OUT: '
+
+   Scenario: Cancel DCB Transaction manually after ITEM_CHECKED_IN
```

Comment:
I think we can merge all the negative scenarios into a single one. Like below.
Once the transaction status is Item_checked_out, we will try to cancel the dcb transaction. We will get 4xx error. Now still the transaction status is ITEM_CHECKED_OUT, so we can use this transaction to updated the status to ITEM_CHECKED_IN and then we will try to cancel the dcb transaction. By this way, we dont need to create multiple scenarios for cancellation negative scenarios.